### PR TITLE
PCHR-1251: Report tab (static)

### PR DESCRIPTION
**Pull request by AkA84**
_(Related to https://github.com/compucorp/civihr-employee-portal-theme/pull/119)_

This PR introduced the first draft of the `<my-leave-report></my-leave-report>` AngularJS component, that contains a static version of the "My Leave - Report" tab. Although the data in it is static, the page is interactive in terms of dropdown and collapsible elements

my-report-leave

# Technical details
## `my-leave-report`
For now this component just holds the information about whether a section is open or not. It will be expanded later on
```js
function controller($log) {
  init.call(this);

  function init() {
    this.isOpen = {
      approved: false,
      entitlement: false,
      holiday: false,
      open: false,
      other: false
    };
  }
}
```

## Collapsible sections
They are achieved simply by using `ng-show`
```html
<tr>
  <td ng-click="report.isOpen.entitlement = !report.isOpen.entitlement">
    <i ng-class="{ 'fa-chevron-down': report.isOpen.entitlement, ' fa-chevron-right': !report.isOpen.entitlement }"></i>
    Period Entitlement
  </td>
  <!-- ... -->
</tr>
<tr ng-show="report.isOpen.entitlement">
  <!-- ... -->
</tr>
```
Initially the idea was to use [`uib-collapse`](https://angular-ui.github.io/bootstrap/#/collapse), but it doesn't work on `<tr>`s and `<td>`. Thus a simpler solution was implemented

## Dropdowns
The dropdown are achieved via the [`uib-dropdown`](https://angular-ui.github.io/bootstrap/#/dropdown) directive
```html
<div uib-dropdown dropdown-append-to-body>
  <a href class="chr_leave-report__action" uib-dropdown-toggle>
    <i class="fa fa-ellipsis-v"></i>
  </a>
  <ul class="dropdown-menu dropdown-menu-right" uib-dropdown-menu>
    <!-- ... -->
  </ul>
</div>
```
The `dropdown-append-to-body` modifier was used as to make sure that the dropdowns could be visualized even when overflowing the nested tables
overflowing-dropdown

### CODE REVIEW

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        $log = _$log_;
+        $rootScope = _$rootScope_;
+
+        spyOn($log, 'debug');
+
+        compileComponent();
+      }));
+
+      it('is initialized', function () {
+        expect($log.debug).toHaveBeenCalled();
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js]: We should not test `$log`

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          return status === false;
+        })).toBe(true);
+      });
+
+      it('is contains the expected markup', function () {
+        expect(component.find('.chr_leave-report').length).toBe(1);
+      });
+
+      function compileComponent() {
+        var $scope = $rootScope.$new();
```
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js]: Put comma between variables. And a single var `statement`

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js (hunk 1 above)
Comment (deb1990) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js]: Instead we can check the isOpen variable.

Comment (AkA84) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js]: I'm not sure about this, you are coupling the initialization of a controller to the presence or not of a variable. What if we decide that we don't need that variable anymore? Or if we decide to remove it?

Calling `$log` is a common pattern in all CiviHR extension, that's why i'm using it to check if the component is initialized

File: uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js (hunk 2 above)
Comment (AkA84) [inline uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/my-leave/components/my-leave-report_test.js]: To be honest this is one of my biggest struggles :D I don't think we settled for a pattern yet! I personally prefer using different `var` statements, but we haven't decided anything yet about it

**Pull request merged**